packages/ui/form/PhoneInput.tsx

```diff
@@ -18,6 +19,36 @@ export type PhoneInputProps = {
 };
 
 function BasePhoneInput({ name, className = "", onChange, value, ...rest }: PhoneInputProps) {
+  // Type guard to check for CountryData type
+  const isCountryData = (data: CountryData | unknown): data is CountryData => {
+    return (data as CountryData).dialCode !== undefined;
+  };
+
+  // This function prefixes dialing code based on country selection.
+  const onCountryChange = (value: string, data: CountryData | unknown) => {
+    let updatedValue = `+${value}`;
+    const phoneNumber = parsePhoneNumberFromString(updatedValue);
+    const isValidPhoneNumber = !!phoneNumber?.isValid();
+    const parsedDialCode = phoneNumber?.countryCallingCode?.toString();
+
+    const selectedDialCode = isCountryData(data) ? data?.dialCode : undefined;
+
+    // Prefix the dialing code only if input value is not a valid phone number,
+    // i.e. if user has not prefixed a dialing code.
+    if (
+      !!value &&
+      !!selectedDialCode &&
+      value !== selectedDialCode &&
+      !!parsedDialCode &&
+      selectedDialCode !== parsedDialCode &&
+      !isValidPhoneNumber
+    ) {
+      updatedValue = `+${selectedDialCode}${value}`;
+    }
```

Comment:
Currently, we don't need to prefix any country code in any case; we want the user to add the full phone number, including the country code.